find_email_addresses.py

```diff
@@ -0,0 +1,183 @@
+from bs4 import BeautifulSoup
+from email.utils import parseaddr
+from urlparse import urlparse
+
+import re
+import requests
+import sys
+
+class Email_Finder():
+	def __init__(self, debug = False, find_all_internal_links = False):
+		# Turns on/off debug mode to print what is happening
+		self.debug = debug
+		# self.find_all_internal_links determines whether the Email Finder should find internal links on all internal pages, or only the homepage
+		# web.mit.edu, for example, is very deep with when searching all internal links
+		self.find_all_internal_links = find_all_internal_links
+		if find_all_internal_links:
+			self.debug_print("Finding all internal links")
+		else:
+			self.debug_print("Finding internal links only from homepage")	
+		self.reset()
+	
+	def reset(self):
+		""" Resets the email Email_Finder
+		"""
+
+		self.debug_print("Resetting Email_Finder")
+		self.all_pages = []
+		self.to_do_pages = []
+		self.all_emails = []
+
+
+	def process_domain(self, domain):
+		""" Processes all the internal links in a domain to find all the email addresses
+		"""
+		
+		self.reset()
+		self.domain = domain
+		# setup the regex used to see if urls are internal
+		self.domain_regex = "//" + self.domain + "|www."+ self.domain
```

Comment:
There might be some issues with this regex. I'm not sure if it's going to handle the case where the original domain is part of the new url. For example:

If the domain is jana.com, and you find a link to jana.comfoo.com I think it will return a match